modelscan/cli.py

```diff
@@ -65,39 +73,67 @@ def cli(
     if log is not None:
         logger.setLevel(getattr(logging, log))
 
-    try:
-        modelscan = Modelscan()
-        if path is not None:
-            pathlibPath = Path().cwd() if path == "." else Path(path).absolute()
-            if not pathlibPath.exists():
-                raise FileNotFoundError(f"Path {path} does not exist")
-            else:
-                modelscan.scan_path(pathlibPath)
-        # elif url is not None:
-        #     modelscan.scan_url(url)
-        elif huggingface is not None:
-            modelscan.scan_huggingface_model(huggingface)
+    modelscan = Modelscan()
+    if path is not None:
+        pathlibPath = Path().cwd() if path == "." else Path(path).absolute()
+        if not pathlibPath.exists():
+            raise FileNotFoundError(f"Path {path} does not exist")
         else:
-            raise click.UsageError(
-                "Command line must include either a path or a Hugging Face model"
-            )
-        ConsoleReport.generate(
-            modelscan.issues,
-            modelscan.errors,
-            modelscan._skipped,
-            show_skipped=show_skipped,
+            modelscan.scan_path(pathlibPath)
+    # elif url is not None:
+    #     modelscan.scan_url(url)
+    elif huggingface is not None:
+        modelscan.scan_huggingface_model(huggingface)
+    else:
+        raise click.UsageError(
+            "Command line must include either a path or a Hugging Face model"
         )
-        return 0
+    ConsoleReport.generate(
+        modelscan.issues,
+        modelscan.errors,
+        modelscan._skipped,
+        show_skipped=show_skipped,
+    )
 
-    except click.UsageError as e:
-        click.echo(e)
-        click.echo(ctx.get_help())
+    # exit code 3 if no supported files were passed
+    if not modelscan.scanned:
+        return 3
+    # exit code 2 if scan encountered errors
+    elif modelscan.errors:
         return 2
+    # exit code 1 if scan completed successfully and vulnerabilities were found
+    elif modelscan.issues.all_issues:
+        return 1
+    # exit code 0 if scan completed successfully and no vulnerabilities were found
+    else:
+        return 0
+
+
+def main() -> None:
+    try:
+        result = cli.main(standalone_mode=False)
+
+    except (
+        click.UsageError,
+        click.BadParameter,
+        click.BadOptionUsage,
+        click.NoSuchOption,
+        click.ClickException,
```

Comment:
It looks like `ClickException` is the parent class of all of the other exceptions so we shouldn't need any of the others.